### ISSUE TITLE
No early video after update to 3.2

### ISSUE THREAD

**Issue by frogblueDFi** [baresip/baresip#2639]
Hi!

I recently updated from baresip version 3.0 to 3.2.
Everything fine but I observed that early video is not functional any more. After answering the call, video is ok.
I checked change "menu limit early video #2514" but can't see what I have to change to get it back working.

Any suggestions? Must I add/change configurations? Additional calls to accept early video?

Regards
Dietmar

**Comment by sreimers** [comment]
Can you test with git main?

This commit may be related: https://github.com/baresip/baresip/commit/8347adaea5696f359cc8a1fb934fcc76979f8e24

**Comment by frogblueDFi** [comment]
Thanx for the quick response!

Sorry, I was wrong, I updated to git main, not 3.2. No early video with git main.
Tried now with git tag v3.2.0, early video is functional!!

Dietmar

**Comment by sreimers** [comment]
CC: @cspiel1 @maximilianfridrich

**Comment by alfredh** [comment]
https://github.com/baresip/baresip/wiki/Guidelines-for-reporting-Issues

please include config and log

**Comment by cspiel1** [comment]
yes, we need config, accounts and log. And the steps how this can be reproduced.

**Comment by frogblueDFi** [comment]
Sorry for not providing config etc.
I use baresip as a library.
Again, v3.2.0 is perfect, main has no early video. Same compilation steps. But only iOS and MacOS. Android not yet available.
Can't provide logs at the moment as I don't have the time today to switch back to main.

Config
------
# SIP
sip_listen		
sip_certificate	
sip_cafile		
sip_capath		
sip_transports		UDP,TCP,TLS,WS,WSS
sip_trans_def	UDP
sip_verify_server			no
sip_tos	160

# Call
call_local_timeout	120
call_max_calls		1
call_hold_other_calls	yes

# Audio
audio_path		
audio_player		coreaudio,
audio_source		coreaudio,
audio_alert		coreaudio,
auplay_srate		16000
ausrc_srate		16000
auplay_channels		1
ausrc_channels		1
audio_txmode		poll
audio_level		no
ausrc_format		S16LE
auplay_format		S16LE
auenc_format		S16LE
audec_format		S16LE
audio_buffer		20-160		# ms
audio_buffer_mode	fixed		# fixed, adaptive
audio_silence		-35.0		# in [dB]
audio_telev_pt		101

# Video
video_source		avcapture,Logitech Webcam C925e
#video_source		avformat,rtmp://127.0.0.1/app/foo
video_display		frogvideo,
video_size		"320x240"
video_bitrate		1000000
video_fps		12.00
video_fullscreen	no
videnc_format		nv12

# AVT
rtp_tos			184
rtp_video_tos		136
rtp_ports		1024-49152
rtp_bandwidth		512-4096
audio_jitter_buffer_type	fixed
audio_jitter_buffer_delay	5-10
video_jitter_buffer_type	fixed
video_jitter_buffer_delay	5-50
rtp_stats		yes
rtp_timeout		0 # in seconds

# Network
net_interface		
net_af			unspecified

Account
--------
"diFiMac <sip:12-dietmar@sip.dev.frogblue.cloud;transport=tls>;auth_pass=<password>regint=60;prio=0;medianat:stun;stunserver=stun:stun.dev.frogblue.cloud:3478"

**Comment by cspiel1** [comment]
Which `answermode` did you set? In the above I could not see that early media is activated. This has to be done in the accounts file at the callee (the UA that receives the call).

**Comment by frogblueDFi** [comment]
We use baresip as lib and set answer mode directly after creating the account.
account_set_answermode(acc, ANSWERMODE_EARLY_VIDEO);
account_set_sip_autoanswer(acc, true);
It's not part of the registration string, but will try it.

**Comment by cspiel1** [comment]
The lib API is also an option. No need to try the same with accounts file.

You have to choose between answermode early video and autoanswer.
- The first leads to answer with 183 Session Progress and enables the early video before the call is established.
- The later establishes the incoming call immediately with a 200 Ok.

Please close the issue if this answers your questions!
